### DISCUSSION TITLE
Specify raneg of data

### DISCUSSION THREAD

**Discussion by suneworld** [EvotecIT/PSWriteHTML#453] [General]
What a wonderful piece of code! Thank you!

My issue is that I have NaN in my data, so it really screws up my graphs. Is it possible to simply disregard (show nothing) if  there is no data?

![image](https://github.com/user-attachments/assets/a726797f-8c2a-4cf8-9e36-e623b90b4456)

**Comment by PrzemyslawKlys** [comment]
How you actually generate it? Whats the code?

**Reply by suneworld** [reply]
`        New-HTMLSection -HeaderText 'Autopilot Enrollment' -TextBackGroundColor "#000000" -HeaderTextSize 22 {
            New-HTMLPanel {
                New-HTMLChart -Title "Autopilot historical" {
                    New-ChartToolbar
                    New-ChartAxisX -Name $labels
                    New-ChartLine -Curve smooth -Width 2 -Name 'Enrolled' -Value $enrolledValues -Color Green
                    New-ChartLine -Curve smooth -Width 2 -Name 'Not Contacted' -Value $notContactedValues -Color RedBerry
                    New-ChartLine -Curve smooth -Width 2 -Name 'Total' -Value $totalAutopilotValues -Color Gray
                    New-ChartAxisY -Show -TitleText 'Number of Devices'
                }
            }

            # Define HTML
            New-HTMLPanel {
                New-HTMLChart -Title "Autopilot Total Registration" {
                    # Get the last value using array indexing
                    New-ChartPie -Name 'Enrolled' -Value ([int]($enrolledValues[-1])) -Color Green
                    New-ChartPie -Name 'Not Contacted' -Value ([int]($notContactedValues[-1])) -Color RedBerry
                }
            }

            New-HTMLPanel {
                New-HTMLChart -Title "Entra Joined Autopilot Registration" {
                    New-ChartPie -Name 'Enrolled' -Value ([int]($enrolledValues[-1])) -Color Green
                    New-ChartPie -Name 'Not Enrolled' -Value ([int]($entraJoinedNotInAP[-1])) -Color RedBerry
                }
            }
        }`

**Comment by suneworld** [comment]
To answer my own question:

I was actually passing values, but they were blank.
To omit blank values:
`  | Where-Object { $_ -ne $null -and $_ -ne "" }`
